pkg/agent/datamodel/sig_config.go

```diff
@@ -896,3 +923,112 @@ func withSubscription(subscriptionID string) SigImageConfigOpt {
 		c.SubscriptionID = subscriptionID
 	}
 }
+
+type Manifest struct {
+	Containerd struct {
+		Edge     string            `json:"edge"`
+		Versions []string          `json:"versions"`
+		Pinned   map[string]string `json:"pinned"`
+	} `json:"containerd"`
+	Runc struct {
+		Versions  []string          `json:"versions"`
+		Pinned    map[string]string `json:"pinned"`
+		Installed map[string]string `json:"installed"`
+	} `json:"runc"`
+	NvidiaContainerRuntime struct {
+		Versions []string `json:"versions"`
+	} `json:"nvidia-container-runtime"`
+	NvidiaDrivers struct {
+		Versions []string `json:"versions"`
+	} `json:"nvidia-drivers"`
+	Kubernetes struct {
+		Versions []string `json:"versions"`
+	} `json:"kubernetes"`
+}
+
+type ProcessedManifest struct {
+	Versions  []string
+	Pinned    map[string]string
+	Edge      string
+	Installed map[string]string
+}
+
+func getCachedVersionsFromManifestJSON(manifestFilePath string) {
```

Comment:
I'd probably prefer this to return the manifest and then within the init() block we actually do the package-level assignment to `CachedFromManifest` - mainly these functions become a bit easier to test independently